internal/batches/executor/run_steps.go

```diff
@@ -38,16 +39,15 @@ type RunStepsOpts struct {
 	Task *Task
 	// TempDir points to where temporary files of the execution should live at.
 	TempDir string
+	// MountDir points to where files that should be mounted live at.
+	MountDir string
```

Comment:
Maybe `WorkspaceDir` describes this better? MountDir makes me think we mount a specific dir.